### ISSUE TITLE
when pre-train discriminator, loss keeps the same

### ISSUE THREAD

**Issue by Slyne** [rickyhan/SimGAN-Captcha#4]
Here i've got some questions:
1.  when pretrain the discriminator, i got a loss of 0.693147 for both real and synthetic images.  And I notice the discriminator predicts both images as [0,0].
Does that mean the discriminator couldn't classify the difference between synthetic images and real images?

2. why we set discriminator_model.trainable = False before pre-training the discriminator ?

**Comment by Slyne** [comment]
Solved.
It's related to the order of model.trainable = False. This parameter should be set before model.compile to make it work.

refiner_model.compile(optimizer=sgd, loss=self_regularization_loss)
discriminator_model.compile(optimizer=sgd, loss=local_adversarial_loss)  #discriminator_model is trainable
discriminator_model.trainable = False  
combined_model.compile(optimizer=sgd, loss=[self_regularization_loss, local_adversarial_loss]) #combined_model cannot update discriminator_model parameters
